### PR TITLE
Miscellaneous `iam.py` docstring updates

### LINKED ISSUE #2250: Miscellaneous `iam.py` docstring updates
Checklist of issues in [`iam.py`](https://github.com/pvlib/pvlib-python/blob/cbe4cc5a8bb3bb529f2dd885aebcd527fedbbc63/pvlib/iam.py) to be revised:

- [x] Incorrect AOI range statement [here](https://github.com/pvlib/pvlib-python/blob/cbe4cc5a8bb3bb529f2dd885aebcd527fedbbc63/pvlib/iam.py#L273)
- [x] Linting (line too long) [here](https://github.com/pvlib/pvlib-python/blob/cbe4cc5a8bb3bb529f2dd885aebcd527fedbbc63/pvlib/iam.py#L894)

Any more?

**Additional Context**
Discussion ([comment](https://github.com/pvlib/pvlib-python/pull/2247#issuecomment-2403462400)) in #2247

Addressed in #2251

### PR DESCRIPTION
**Pull request by RDaxini**
- [X] Closes #2250
 - [X] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [ ] ~Tests added~
 - [ ] ~Updates entries in [`docs/sphinx/source/reference`](https://github.com/pvlib/pvlib-python/blob/main/docs/sphinx/source/reference) for API changes.~
 - [ ] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/main/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [X] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

This PR resolves issues listed in #2250

### CODE REVIEW

File: pvlib/iam.py [hunk 1]
```diff
@@ -292,7 +292,6 @@ def martin_ruiz(aoi, a_r=0.16):
     pvlib.iam.interp
     pvlib.iam.sapm
     '''
-    # Contributed by Anton Driesse (@adriesse), PV Performance Labs. July, 2019
```
Comment (kandersolar) [inline pvlib/iam.py:295]: I disagree that this was a copy/paste error; it was included in the original PR contributing this function: #752

Comment (cwhanse) [reply pvlib/iam.py:295]: I don't know how that line was inserted at the head of the `martin_ruiz` function, as that wasn't contributed by Anton, but may have been edited to apply some processing that was copied from `martin_ruiz_diffuse` or other functions that Anton contributed.

Comment (kandersolar) [reply pvlib/iam.py:295]: Perhaps I am misunderstanding something, but (ignoring irrelevant shuffling between modules) `martin_ruiz` was originally contributed in #752, which was indeed from Anton.

Comment (cwhanse) [reply pvlib/iam.py:295]: @kandersolar is correct, I didn't follow the blame history to the origin.

The comment should be restored.

File: pvlib/iam.py [hunk 2]
```diff
@@ -270,7 +270,7 @@ def martin_ruiz(aoi, a_r=0.16):
     which is presented as :math:`AL(\alpha) = 1 - IAM` in equation 4 of [1]_,
     with :math:`\alpha` representing the angle of incidence AOI. Thus IAM = 1
     at AOI = 0, and IAM = 0 at AOI = 90.  This equation is only valid for
```
Comment (echedey-ls) [inline pvlib/iam.py:272]:
```suggestion
    at AOI = 0°, and IAM = 0 at AOI = 90°.  This equation is only valid for
```

File: pvlib/iam.py [hunk 3]
```diff
@@ -270,7 +270,7 @@ def martin_ruiz(aoi, a_r=0.16):
...
-    -90 <= aoi <= 90, therefore `iam` is constrained to 0.0 outside this
+    0 <= aoi <= 90, therefore `iam` is constrained to 0.0 outside this
```
Comment (echedey-ls) [inline pvlib/iam.py:273]:
```suggestion
    0° <= aoi <= 90°, therefore `iam` is constrained to 0.0 outside this
```

**Review by echedey-ls — COMMENTED** [review]
LGTM! It's a pleasure to announce that `iam.py` now is PEP8 compliant :)

**Comment by RDaxini** [comment]
Do we need a whatsnew entry?

**Review by kandersolar — APPROVED** [review]
Thanks @RDaxini and @echedey-ls!

**Pull request merged**